### PR TITLE
THORN-2424: Update to smallrye-jwt-1.1-2.0.0 version

### PR DESCRIPTION
**Pull request by sberyozkin**
Motivation
----------
Update to the latest smallrye-jwt-1.1 version

Modifications
-------------
smallrye-jwt-1.1 has the original Thorntail MP JWT JAX-RS filters incorporated (with some minor updates) - this let remove the filters from Thorntail; it also encapsulates the token extraction logic so `JWTAuthenticationMechanism` only needs to provide the header values.
Additionally, as part of this PR, THORN-2408 (support Thorntail specific properties for configuring token headers and default groups) and THORN-2427 (verify that Keycloak tokens can be validated directly with microprofile-jwt, using Thorntail specific properties) are addressed.

Result
------
Less code in the 'microprofile-jwt', better support for 3rd party tokens

### CODE REVIEW

**Comment by sberyozkin** [comment]
Hi @Ladicek have a look please when you get some time for an initial review; I'll update the PR subject once the next smallrye-jwt-1.1 is released

**Comment by Ladicek** [comment]
I'll take a look, but next time, you can use https://github.blog/2019-02-14-introducing-draft-pull-requests/ to signify that the PR is not to be merged yet :-)

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/MicroProfileJWTAuthFraction.java [hunk 1]
```diff
@@ -62,6 +62,20 @@
...
+    @Configurable("thorntail.microprofile.jwt.token.header")
+    private Defaultable<String> tokenHeader = string("Authorization");
+    @AttributeDocumentation("Cookie name containing a JWT token. This property is ignored unless the 'thorntail.microprofile.jwt.token.header' is set to 'Cookie'")
+    @Configurable("thorntail.microprofile.jwt.token.cookie")
+    private String tokenCookie;
+    @AttributeDocumentation("Default group name. This property can be used to support the JWT tokens without a 'groups' claim.")
+    @Configurable("thorntail.microprofile.jwt.claims.groups")
+    private String defaultGroupsClaim;
+
+    @AttributeDocumentation("JSON path to the claim containing an array of groups. It can be used a token has no 'groups' claim but has the groups set in a different claim")
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/MicroProfileJWTAuthFraction.java]: It's not "JSON Path". The syntax is actually closer to JSON Pointer, but it's not it either. Avoiding "JSON" + adding an example would be best I think.

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/MicroProfileJWTAuthFraction.java [hunk 2]
```diff
@@ -62,6 +62,20 @@
...
 
+    @AttributeDocumentation("HTTP header which is expected to contain a JWT token, default value is 'Authorization'")
+    @Configurable("thorntail.microprofile.jwt.token.header")
+    private Defaultable<String> tokenHeader = string("Authorization");
+    @AttributeDocumentation("Cookie name containing a JWT token. This property is ignored unless the 'thorntail.microprofile.jwt.token.header' is set to 'Cookie'")
+    @Configurable("thorntail.microprofile.jwt.token.cookie")
+    private String tokenCookie;
+    @AttributeDocumentation("Default group name. This property can be used to support the JWT tokens without a 'groups' claim.")
+    @Configurable("thorntail.microprofile.jwt.claims.groups")
+    private String defaultGroupsClaim;
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/MicroProfileJWTAuthFraction.java:75]: Empty lines between individual fields would be good :-)

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/JWTAuthMechanism.java [hunk 3]
```diff
@@ -154,4 +130,23 @@ protected RoleGroup extract(Subject subject) {
         RoleGroup roles = new SimpleRoleGroup(rolesGroup);
         return roles;
     }
+
+    private static class UndertowBearerTokenExtractor extends AbstractBearerTokenExtractor {
+        private HttpServerExchange httpExchange;
+        UndertowBearerTokenExtractor(JWTAuthContextInfo authContextInfo, HttpServerExchange exchange) {
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/JWTAuthMechanism.java:137]: Empty line between field and ctor please :-)

**Comment by Ladicek** [comment]
Otherwise looking good, thanks!

**Comment by sberyozkin** [comment]
Sure, will try a draft PR next time

**Comment by sberyozkin** [comment]
@Ladicek This is now green, please check again next week

**Review by Ladicek — APPROVED** [review]

**Comment by Ladicek** [comment]
Nice, thanks!

**Pull request merged**